src/pages/commits/ActiveCommits/BuildVariantCard.test.tsx

```diff
@@ -0,0 +1,78 @@
+import { MockedProvider } from "@apollo/client/testing";
+import {
+  renderWithRouterMatch as render,
+  screen,
+  userEvent,
+  waitFor,
+} from "test_utils";
+import { BuildVariantCard } from "./BuildVariantCard";
+import { injectGlobalDimStyle, removeGlobalDimStyle } from "./utils";
+
+jest.mock("./utils");
+
+describe("buildVariantCard", () => {
+  it("should call the appropriate functions on hover and unhover", async () => {
+    const mockIntersectionObserver = jest.fn((callback) => {
+      callback([
+        {
+          isIntersecting: true,
+        },
+      ]);
+      return {
+        observe: jest.fn(),
+        disconnect: jest.fn(),
+      };
+    });
+
+    // @ts-expect-error
+    window.IntersectionObserver = mockIntersectionObserver;
```

Comment:
If you want to avoid `@ts-expect-error`, I think this is also an option:
```suggestion
     Object.defineProperty(window, "IntersectionObserver", {
         writable: true,
         value: mockIntersectionObserver,
    });
```